packages/testing/src/testing.ts

```diff
@@ -106,7 +106,10 @@ export function createTestingPinia({
   fakeApp = false,
   createSpy: _createSpy,
 }: TestingOptions = {}): TestingPinia {
-  const pinia = createPinia()
+  const app: App = (globalThis as any).__unctx__.get('nuxt-app').tryUse().vueApp
+  const symbols = Object.getOwnPropertySymbols(app._context.provides)
+  const nuxtPinia: Pinia = app._context.provides[symbols[0]]
+  const pinia = nuxtPinia ?? createPinia()
```

Comment:
Thanks but not only we **always** want to create a new pinia instance here, this code is relying on internals and globals that are specific to nuxt. This will break in any other scenario

Reply:
This is temporary fix code. You are right, this code lacks flexibility. However, there is another issue where `addPlugin` from `@nuxt/pinia` creates a new pinia instance, which conflicts with the creation of `createTestingPinia`. But the logic of `@nuxt/pinia` is reasonable, and I believe it should not be modified.

Reply:
It's okay to adapt `@nuxt/pinia` to work with the test environment. It's really crucial to always create a new pinia instance

Reply:
got it.